stdlib/src/os/os.mojo

```diff
@@ -328,3 +328,60 @@ fn unlink[pathlike: os.PathLike](path: pathlike) raises:
 
     """
     remove(path.__fspath__())
+
+
+# ===----------------------------------------------------------------------=== #
+# mkdir/rmdir
+# ===----------------------------------------------------------------------=== #
+
+
+fn mkdir(path: String, mode: Int = 0o777) raises:
+    """Creates a directory at the specified path.
+    If the directory can not be created an error is raised.
+    Absolute and relative paths are allowed, relative paths are resolved from cwd.
+
+    Args:
+      path: The path to the directory.
+      mode: The mode to create the directory with.
+    """
+
+    var error = external_call["mkdir", Int](path._as_ptr(), mode)
```

Comment:
The result should be an int32 instead of an Int. An Int in mojo corresponds to a ssize_t